CMakeLists.txt

```diff
@@ -154,6 +154,11 @@ string(REGEX MATCH "Clang" CMAKE_COMPILER_IS_CLANG "${CMAKE_C_COMPILER_ID}")
 
 include(CheckCCompilerFlag)
 
+if (NOT CMAKE_VERSION VERSION_LESS 3.1.0)
+    set(CMAKE_C_STANDARD 99)
```

Comment:
Is it a problem to not use this at all (remove those 4 lines) and to do "set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")" unconditionally for GNU and ARM compiler?

Reply:
add `-std=c99` to `CMAKE_C_FLAGS` for GNU and ARM compiler is simple.
but `CMAKE_C_STANDARD` and `CMAKE_C_EXTENSIONS` is standard variables in CMake.
although they are exists after CMake 3.1.0
adding `-std=c99` and some other compiler-specific script should be removed eventually after we drop support for old CMake version

Reply:
I haven't followed all the conversation so I'm not sure if this is relevant. For information, we try to support maintained versions of Ubuntu and CentOS, which currently means [CMake back to 2.8.12.2](https://github.com/ARMmbed/mbedtls/issues/2896#issuecomment-603471273). And [armcc doesn't support `-std`](https://github.com/ARMmbed/mbedtls/pull/3641#discussion_r494993625).

Reply:
ok, I am convinced with your arguments, let's keep the setting of CMAKE_C_STANDARD. Rather than:
if (NOT CMAKE_VERSION VERSION_LESS 3.1.0), I would prefer:
if (CMAKE_VERSION VERSION_GREATER ... ).

Reply:
we need a expression `cmake >= 3.1.0`.
but `VERSION_GREATER_EQUAL` cmake introduced in cmake 3.7.
so we have to use `not NOT CMAKE_VERSION VERSION_LESS 3.1.0`